packages/hyperion-autologging/src/ALSurfaceVisibilityPublisher.ts

```diff
@@ -97,11 +102,21 @@ export function publish(options: InitOptions): void {
             observer.unobserve(element);
             observedRoots.delete(element);
           }
+          if (!optimizeSurfaceMaps) {
+            activeSurfaces.delete(event.surface);
+          }
         }
         break;
       }
     }
 
+    if (!optimizeSurfaceMaps) {
+      // We know this is horrible! But keeping here to compare the performance of it for now
+      tmpInit();
```

Comment:
This is the main portion you are testing removing?

Reply:
Yep! It's crazy we missed this for so long!